src/configobject.h

```diff
@@ -34,23 +35,32 @@
 
 class ConfigKey {
   public:
-    ConfigKey();
+    ConfigKey(); // is required for qMetaTypeConstructHelper()
+    ConfigKey(const ConfigKey& key);
     ConfigKey(const QString& g, const QString& i);
-    ConfigKey(const char* g, const char* i);
     static ConfigKey parseCommaSeparated(QString key);
 
     inline bool isNull() const {
         return group.isNull() && item.isNull();
     }
 
+    // comparison function for ConfigKeys. Used by a QHash in ControlObject
+    inline bool operator==(const ConfigKey& other) const {
```

Comment:
Our style guide gives a reason why to prefer it: "Prefer to define non-modifying binary operators as non-member functions. If a binary operator is defined as a class member, implicit conversions will apply to the right-hand argument, but not the left-hand one. It will confuse your users if a < b compiles but b < a doesn't."

https://google.github.io/styleguide/cppguide.html#Operator_Overloading